src/tool/GenerateAction.cpp

```diff
@@ -13,15 +13,108 @@
 #include "lib/Lib/AbsoluteCompilationDatabase.hpp"
 #include "lib/Lib/ConfigImpl.hpp"
 #include "lib/Lib/CorpusImpl.hpp"
+#include "llvm/Support/Program.h"
 #include <mrdocs/Generators.hpp>
 #include <mrdocs/Support/Error.hpp>
 #include <mrdocs/Support/Path.hpp>
 #include <clang/Tooling/JSONCompilationDatabase.h>
+
 #include <cstdlib>
+#include <unordered_map>
 
 namespace clang {
 namespace mrdocs {
 
+std::optional<std::string> 
+getCompilerInfo(llvm::StringRef compiler) 
```

Comment:
Does the result of `c++ -v -E -x c++ -` have a more formal name than compiler info? I have the impression this name will be taken soon by a function that returns a complete `struct`.

Reply:
there's no formal name for the combination of options used in the g++ -v -E -x c++ - command. It's more of a commonly used trick to extract detailed compiler configuration information. I will change the name of the function with a better name.

Reply:
I imagined the documentation of the `-v -E -x c++` flags would have some name we could use.